### PR TITLE
[coremedia] Modernize types

### PR DESCRIPTION
**Pull request by spouliot**
* Use `is` instead of `==` for null checks
* Use `ThrowHelper` to throw common exceptions
* Use faster `CFString` API (over the slower `NSString` variants)
* Use `HashCode.Combine` for net profile

### CODE REVIEW

**Review by tj-devel709 — APPROVED** [review]

File: src/CoreMedia/CMFormatDescription.cs [hunk 1]
```diff
@@ -489,13 +489,13 @@ public bool VideoMatchesImageBuffer (CVImageBuffer imageBuffer)
 		public static CMVideoFormatDescription? FromHevcParameterSets (List<byte[]> parameterSets, int nalUnitHeaderLength, NSDictionary extensions, out CMFormatDescriptionError error)
 		{
 			if (parameterSets is null)
-				throw new ArgumentNullException (nameof (parameterSets));
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (parameterSets));
 
 			if (parameterSets.Count < 3)
-				throw new ArgumentException ($"{nameof (parameterSets)} must contain at least three elements");
+				ObjCRuntime.ThrowHelper.ThrowArgumentException ("must contain at least three elements", nameof (parameterSets));
```
Comment (rolfbjarne) [inline src/CoreMedia/CMFormatDescription.cs:495]: Argument name comes first, then the message:

```suggestion
				ObjCRuntime.ThrowHelper.ThrowArgumentException (nameof (parameterSets), "must contain at least three elements");
```

File: src/CoreMedia/CMSampleBuffer.cs [hunk 2]
```diff
@@ -745,9 +745,9 @@ public CMSampleBufferError CopyPCMDataIntoAudioBufferList (int frameOffset, int
 			CMFormatDescription formatDescription, CMSampleTimingInfo[] sampleTiming, out CMSampleBufferError error)
 		{
 			if (sampleTiming is null)
-				throw new ArgumentNullException (nameof (sampleTiming));
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (sampleTiming));
 			if (sampleTiming.Length != 1)
-				throw new ArgumentException ("Only a single sample is allowed.", nameof (sampleTiming));
+				ObjCRuntime.ThrowHelper.ThrowArgumentException ("Only a single sample is allowed.", nameof (sampleTiming));
```
Comment (rolfbjarne) [inline src/CoreMedia/CMSampleBuffer.cs:750]: Same

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**